Source/Plugin.BLE/Android/Adapter.cs

```diff
@@ -304,7 +304,7 @@ public override async Task<IDevice> ConnectToKnownDeviceNativeAsync(Guid deviceG
         {
             var macBytes = deviceGuid.ToByteArray().Skip(10).Take(6).ToArray();
             var nativeDevice = _bluetoothAdapter.GetRemoteDevice(macBytes);
-            if (nativeDevice == null)
+            if (nativeDevice == null || !(BluetoothDeviceType.Le == nativeDevice.Type || BluetoothDeviceType.Dual == nativeDevice.Type))
                 throw new Abstractions.Exceptions.DeviceConnectionException(deviceGuid,"", $"[Adapter] Device {deviceGuid} not found.");
             var device = new Device(this, nativeDevice, null);
```

Comment:
Thanks for the contribution! Looks pretty good :)

Two minor comments/suggestions:
1. How about introducing a small auxiliary method in the `Adapter` class? (Apart from this place, it could also be used in `GetSystemConnectedOrPairedDevices` and `GetBondedDevices`.)
```
        static bool SupportsBLE(BluetoothDevice d)
        {
            return d.Type == BluetoothDeviceType.Le || d.Type == BluetoothDeviceType.Dual;
        }
```
2. I think the error message in the exception is not fully accurate for the new case you're adding (the device with the given GUID exists, but just does not support BLE). It might be better to generate an exception with a slightly different error message. Maybe something like this: "[Adapter] Device {deviceGuid} does not support BLE."